public/app/plugins/datasource/elasticsearch/components/QueryEditor/MetricAggregationsEditor/MetricEditor.tsx

```diff
@@ -69,6 +69,11 @@ export const MetricEditor = ({ value }: Props) => {
   const dispatch = useDispatch();
   const getFields = useFields(value.type);
 
+  const getTypeOptionsAsync = async (previousMetrics: MetricAggregation[], xpack = false) => {
```

Comment:
I think it would be worth fetching ES version in `init` in data source. For me, it seems that `init` would be perfect place for it and then we wouldn't have to have all of these async calls. 

I understand that we are trying to only check version when it is needed, but I am wondering if 1 call during `init` phase would be really wasteful? 

Moreover, we lost showing of "this version of ES is not supported anymore" and if we checked for version in init, we could still show it.

Reply:
good idea! i checked it, but unfortunately it does not help much. problem is, `init()` is sync. ( https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/datasource.ts#L248 ),

so, i can start loading the db-version in `init()`, but when init returns, the info is not available yet.
so for now i did not implement `init()`.

if you'd prefer so, i can certainly add the `init()`, where it will start loading the db-info, so when later someone else calls getDbInfo, it will appear faster, no problem.  WDYT?